src/net.c

```diff
@@ -827,10 +837,12 @@ static int hyper_check_device_match_mac_addr(const char *device,
 		goto err;
 	}
 
+	free(tmp_mac_addr);
 	close(sock);
 	return 0;
```

Comment:
Yes it would work the same, I just don't know what is preferred in this kind of case. @jodh-intel do you agree with @amshinde about this syntax ?